### PR TITLE
[13.0][ADD] + [MIG] agreement_rebate

### PR DESCRIPTION
**Pull request by sergio-teruel**
cc @Tecnativa TT27463

ping @carlosdauden @chienandalu 

This PR is a migration from this PR (https://github.com/OCA/contract/pull/510)
The v12.0 PR will be closed.

### CODE REVIEW

**Review by carlosdauden — APPROVED** [review]

File: agreement_rebate/README.rst [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+.. contents::
+   :local:
+
+Usage
+=====
+
+To use this module you need to:
+
+#. Go to a *Agreements > agreement*.
```
Comment (chienandalu) [inline agreement_rebate/README.rst:41]: I don't really know what to test :thinking:

**Review by chienandalu — APPROVED** [review]

**Comment by pedrobaeza** [comment]
@sergio-teruel can you please squash commit history a bit?

File: agreement_rebate/wizards/invoice_create.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    def action_create_invoice(self):
+        self.ensure_one()
+        settlements = self.env["agreement.rebate.settlement"].search(
+            self._prepare_settlement_domain()
+        )
+        settlements -= settlements.filtered(
+            lambda s: any(
+                ail.move_id.type == self.invoice_type
+                for ail in s.line_ids.mapped("invoice_line_ids").filtered(
+                    lambda ln: ln.state == "invoiced"
```
Comment (carlosdauden) [inline agreement_rebate/wizards/invoice_create.py:92]: Do you want filter settlement with state invoiced or account.move.line with parent_state (draft|post)?
state field don't exists in account.move.line

**Comment by sergio-teruel** [comment]
@pedrobaeza I have a one commit only!

File: agreement_rebate/README.rst (hunk 1 above)
Comment (sergio-teruel) [reply agreement_rebate/README.rst:41]: Sorry @chienandalu I already complete readme

**Comment by rousseldenis** [comment]
@sergio-teruel Is this ready ?

**Comment by sergio-teruel** [comment]
yes @rousseldenis

**Pull request merged**